### ISSUE TITLE
Question about GROP implementation: missing clipping and min-reduction?

### ISSUE THREAD

**Issue by liu-s-q19** [huggingface/open-r1#227]
Hi,
I was going through the implementation of GROP in your repository and noticed that the loss function is implemented as:
```python
per_token_loss = torch.exp(per_token_logps - per_token_logps.detach()) * advantages.unsqueeze(1)
per_token_loss = -(per_token_loss - self.beta * per_token_kl)
```
However, in the paper, the loss function includes clipping and taking the minimum, which I don't see here. 
![Image](https://github.com/user-attachments/assets/36f6a376-612f-4a3a-ac31-c411b1198d8c)
Could you clarify:
1. Why is clipping not included in the implementation?
2. How does this affect the optimization compared to the approach in the paper?
3. If I wanted to add clipping and min-reduction, how should I modify this implementation?

Thanks in advance for your help!

**Comment by qgallouedec** [comment]
See https://github.com/huggingface/trl/issues/2608
